examples/CommonGateway/c-sensors/readme.md

```diff
@@ -1 +1,129 @@
+            uint32_t humidity = (sensorEnvironmentData.humidity * 1000) / 1024;
+#ifdef BME280_64BIT_ENABLE
+            // If macro "BME280_64BIT_ENABLE" is enabled, which it is by default, the unit is 100 * Pascal
+            uint32_t pressure = sensorEnvironmentData.pressure;
+#else
+            // If this macro is disabled, Then the unit is in Pascal
+            uint32_t pressure = sensorEnvironmentData.pressure * 100;
+#endif
+            // clang-format off
+            LOG_DEBUG("Temperature: %ld.%02lu;\tPresure: %lu.%03luhPa;\tHumidity: %lu.%03lu%%RH",
+                    temperature / 100,
+                    temperature % 100,
+                    pressure / 10000,
+                    (pressure % 10000) / 10,
+                    humidity / 1000,
+                    humidity % 1000);
+            // clang-format on
+#endif
+        }
+        else
+        {
+            LOG_DEBUG("Env sensor read failed!\r\n");
+        }
+    }
+
+    if (RETCODE_OK == returnValue)
+    {
+        returnValue = SensorAccelerometer_Read(&sensorAccelerometerData);
+        if (RETCODE_OK == returnValue)
+        {
+            // clang-format off
+            LOG_DEBUG("X: %ld;\tY: %ld;\tZ: %ld",
+                    sensorAccelerometerData.x,
+                    sensorAccelerometerData.y,
+                    sensorAccelerometerData.z);
+            // clang-format on
+        }
+        else
+        {
+            LOG_DEBUG("Accel sensor read failed!\r\n");
+        }
+    }
+
+    if (RETCODE_OK == returnValue)
+    {
+        BSP_LED_Switch(COMMONGATEWAY_LED_GREEN_ID, COMMONGATEWAY_LED_COMMAND_TOGGLE);
+    }
+    else
+    {
+        BSP_LED_Switch(COMMONGATEWAY_LED_GREEN_ID, COMMONGATEWAY_LED_COMMAND_OFF);
+        BSP_LED_Switch(COMMONGATEWAY_LED_RED_ID, COMMONGATEWAY_LED_COMMAND_ON);
+    }
+
+    if (RETCODE_OK == returnValue)
+    {
+        BSP_Board_Delay(5);
+        returnValue = CmdProcessor_Enqueue((CmdProcessor_T *)CmdProcessorHandle, appPeriodicTest, CmdProcessorHandle, UINT32_C(0));
+    }
+}
+```
+This code through in two main functions to get the output.
+The first one is for the Enviroment Sensor, which go through two functions called [SensorEnvironment_Init](https://github.com/eclipse/kiso/blob/master/examples/CommonGateway/c-sensors/source/env_sensor.c#L58) and [SensorEnvironment_Read](https://github.com/eclipse/kiso/blob/master/examples/CommonGateway/c-sensors/source/env_sensor.c#L110) as you can see that these two functions give the measurements continuously for the (temperature, pressure and humidity).
+The second one is for Accelerometer Sensor, which go through two functions called [SensorAccelerometer_Init](https://github.com/eclipse/kiso/blob/master/examples/CommonGateway/c-sensors/source/accel_sensor.c#L48) and [SensorAccelerometer_Read](https://github.com/eclipse/kiso/blob/master/examples/CommonGateway/c-sensors/source/accel_sensor.c#L76) as you can see that these two functions give the dimensions (x,y,z).
```

Comment:
```suggestion
The app is structured in three modules:

* the main app logic
* environment sensor and
* acceleration sensor logic.

With the main app logic being at top of the hierarchy.

Each module has to be initialized first, before it can be used. The app logic initializes vital system components such as the sensor BSP components, as well as the I2C communication bus they are sit on. Finally each sensor module initializes the `bma2x2.h` and `bme280.h` library modules respectively.

Once initialization is complete, the main application loop is entered. This is done using the main command processor, created in `main()`. For each iteration of the app loop, `appPeriodicTest()` is enqueued at the end of the last iteration. This is repeated endlessly.

Reading acceleration data from `SensorAccelerometer_Read()`, one can directly access the x, y and z components as provided by the sensor. Environmental data however, need some static temperature correction first. This compensation can be changed via the `DEFAULT_TEMPERATURE_OFFSET` constant, with the measured temperature offset at a known temperature.
```